services/service-core/src/app.module.ts

```diff
@@ -55,8 +55,8 @@ import { configuration } from './config/config.service';
       imports: [ConfigModule],
       useFactory: async (configService: ConfigService) => ({
         authServerUrl: configService.get('keycloakServerUrl'),
-        realm: configService.get('keycloak.realm'),
-        resource: configService.get('keycloak.nestClientId'),
+        realm: configService.get('keycloak').realm,
+        resource: configService.get('keycloak').nestClientId,
         secret: configService.get('KEYCLOAK_SECRET'),
```

Comment:
Again same comment here since your config is like 
```
"keycloak": {
    "keycloakServerUrl": "",
    "realm": "",
    "nestClientId": "",
    "realmUrl": "",
    "adminUrl": "",
    "grantType": "",
    "clientId": ""
  }
``` 
I can't see any reason why it can't be done this way
```typescript
useFactory: async (configService: ConfigService) => {
        const { authServerUrl, realm, nestClientId, secret } = configService.get('keycloak');
        return {
          authServerUrl,
          realm,
          resource: nestClientId,
          secret,
          'public-client': true,
          verifyTokenAudience: true,
          'confidential-port': 0
        };
      },
```

Reply:
it is inside decorator and you couldn't destruct object inside their, I think